File: scratch/extract_forms.py
import os
import sys

# Ensure pypdf is installed
try:
    import pypdf
except ImportError:
    import subprocess
    subprocess.run([sys.executable, "-m", "pip", "install", "pypdf"])
    import pypdf

folder = r"c:\Users\elmig\Documents\MCC IA\Vila de Fenals\materia_huespedes\HM582Y32HZ"
files = [
    "Dylan - Registration Form - firmado.pdf",
    "Gijs - Registration Form - firmado.pdf",
    "Siard - Registration Form - firmado.pdf",
    "Wolter - Registration Form - firmado.pdf"
]

for name in files:
    path = os.path.join(folder, name)
    print(f"=== FILE: {name} ===")
    try:
        reader = pypdf.PdfReader(path)
        text = ""
        for i, page in enumerate(reader.pages):
            text += page.extract_text() + "\n"
        print(text)
    except Exception as e:
        print(f"Error reading {name}: {e}")
    print("-" * 50)

File: scratch/extract_forms.js
global.DOMMatrix = class DOMMatrix {};
const fs = require('fs');
const path = require('path');
const { PDFParse } = require('pdf-parse');

const targetFolder = "c:/Users/elmig/Documents/MCC IA/Vila de Fenals/materia_huespedes/HM582Y32HZ";
const files = [
  "Dylan - Registration Form - firmado.pdf",
  "Gijs - Registration Form - firmado.pdf",
  "Siard - Registration Form - firmado.pdf",
  "Wolter - Registration Form - firmado.pdf"
];

async function extract() {
  for (const file of files) {
    const filePath = path.join(targetFolder, file);
    console.log(`=== FILE: ${file} ===`);
    try {
      const dataBuffer = fs.readFileSync(filePath);
      const uint8Array = new Uint8Array(dataBuffer);
      const pdfInstance = new PDFParse(uint8Array);
      const data = await pdfInstance.getText();
      console.log(data.text || data);
    } catch (err) {
      console.error(`Error reading ${file}:`, err.message);
    }
    console.log("-".repeat(50));
  }
}

extract();
